### PR TITLE
Add Racket

### PR DESCRIPTION
**Pull request by jryans**
This adds an initial version of the Racket toolchain.

Racket language support PR: https://github.com/compiler-explorer/compiler-explorer/pull/4098

### CODE REVIEW

File: bin/yaml/racket.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  racket:
+    type: script
+    check_exe: bin/racket --version
+    dir: racket-{name}
+    fetch:
+      - https://download.racket-lang.org/releases/{name}/installers/racket-{name}-x86_64-linux-cs.sh racket-{name}.sh
+    script: |
+      sh racket-{name}.sh --in-place --dest ./racket-{name}
+    after_stage_script:
+      - bin/raco pkg install -i --auto disassemble
```
Comment (RubenRBS) [inline bin/yaml/racket.yaml:11]: And this? Does it also keep everything in the final compiler folder?

**Review by RubenRBS — COMMENTED** [review]
Some questions as it might not work out of the box for our setup without some modifications :)

File: bin/yaml/racket.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+compilers:
+  racket:
+    type: script
+    check_exe: bin/racket --version
+    dir: racket-{name}
+    fetch:
+      - https://download.racket-lang.org/releases/{name}/installers/racket-{name}-x86_64-linux-cs.sh racket-{name}.sh
+    script: |
+      sh racket-{name}.sh --in-place --dest ./racket-{name}
```
[the remark this answers is not in the document]
Comment (jryans) [reply bin/yaml/racket.yaml:9]: The file is a *.tar.gz archive with a shell script attached to offer a few user options. The script won't download anything extra, as the data is all attached to end of the script. It depends on access to `gunzip` and `tar` to extract itself.

As for where it puts things, all files are placed in the `--dest` folder. It's a typical Unix-like program directory, with `bin`, `lib`, subdirectories.

File: bin/yaml/racket.yaml (hunk 1 above)
Comment (jryans) [reply bin/yaml/racket.yaml:11]: Yes, this is also self-contained. The `-i` argument means "installation scope", which equates to the `share/pkgs` subdirectory of the compiler folder.

File: bin/yaml/racket.yaml (hunk 2 above)
Comment (RubenRBS) [inline bin/yaml/racket.yaml:9]: I forgot we're trying to get everything into subdirectories, so something like this:

```suggestion
    dir: racket/racket-{name}
    fetch:
      - https://download.racket-lang.org/releases/{name}/installers/racket-{name}-x86_64-linux-cs.sh racket-{name}.sh
    script: |
      sh racket-{name}.sh --in-place --dest ./racket/racket-{name}
```

(Note that it also needs an update in the main repo PR to match the new paths)

Comment (jryans) [reply bin/yaml/racket.yaml]: Ah, makes sense, I've updated this to match.

**Comment by jryans** [comment]
@RubenRBS I've added the extra language-level directory. I also merged the after stage script into the main script, since the script type doesn't support the after stage feature. (I had initially tested the package installation step with a different type that did support after stage.)

**Comment by RubenRBS** [comment]
Nice catch! This looks good now, we only need to test that it actually works in the live site. I'll try to get it merged this weekend :)

**Review by RubenRBS — APPROVED** [review]

**Comment by mattgodbolt** [comment]
I'm happy to merge this and test now!

**Comment by mattgodbolt** [comment]
Seems to have installed just fine:
```
Oct/01 16:18 admin-node~/infra (main|✔) $ /opt/compiler-explorer/racket/racket-8.6/bin/racket --version
Welcome to Racket v8.6 [cs].
```

**Comment by jryans** [comment]
Thanks for the merge, looks good to me! 😄

**Pull request merged**